internal/controller/controllers/agent_controller.go

```diff
@@ -744,14 +744,29 @@ func (r *AgentReconciler) deregisterHostIfNeeded(ctx context.Context, log logrus
 // * No BMH exists for agent
 func (r *AgentReconciler) shouldApproveCSRsForAgent(ctx context.Context, agent *aiv1beta1.Agent, h *models.Host) (bool, error) {
 	if funk.Contains([]models.HostStage{models.HostStageRebooting, models.HostStageJoined}, h.Progress.CurrentStage) {
-		isNone, err := isAgentInNonePlatformCluster(ctx, r.Client, agent)
+		cd, err := getClusterDeploymentFromAgent(ctx, r.Client, agent)
 		if err != nil {
 			return false, err
 		}
+
+		isNone, _, err := isNonePlatformCluster(ctx, r.Client, cd)
 		if isNone {
 			return true, nil
 		}
 
+		if err != nil {
```

Comment:
Should we check this error before we do the check above?
This seems like it should happen before, just in case there was an error?

Reply:
Yes, good catch, thanks Paul!